### PR TITLE
Added the getAllFactorizations(nb: number): number[][] method and unit tests covering its' functionality

### PR DESCRIPTION
**Pull request by dejanko25**
I have written the implementation for both positive and negative numbers.

### CODE REVIEW

**Review by yactouat — CHANGES_REQUESTED** [review]
hey @dejanko25 , thanks again for this wonderful work ! I left a few remarks for you to check ;) please dont hesitate to get back to me if you disagree

**Comment by dejanko25** [comment]
Thank you very much for the feedback @yactouat , those are good remarks. I must admit the ~~ rounding is uncommon, and adding an example of output is a great idea!

I have implemented the changes, wrote a few more test cases for odd positive and negative numbers, and 2 test cases to make sure the exception throwing for decimal numbers works - everything seems fine :)

File: src/index.ts [hunk 1]
```diff
@@ -250,3 +250,59 @@ export const isPrime = (nb: number): boolean => {
   }
   return false;
 };
+
+/**
+ * returns all combinations of factors for the given number
+ *
+ * but [n1, n2] = [n2, n1] and only one is included
+ * consider all factors generated in the same order as below example for 64
+ * [ [64, 1], [ 32, 2 ], [ 16, 4 ], [ 8, 8 ], [ 4, 16 ], [ 2, 32 ], [ 1, 64 ] ]
```
Comment (yactouat) [inline src/index.ts:259]: thank you very much for this

**Review by yactouat — APPROVED** [review]
